docs/data-type-support.rst

```diff
@@ -0,0 +1,79 @@
+.. meta::
+   :description: hipcub API library data type support
+   :keywords: hipcub, ROCm, API library, API reference, data type, support
+
+.. _data-type-support:
+
+******************************************
+Data type support
+******************************************
+
+* Supported input and output types.
+
+  .. list-table:: Supported Input/Output Types
+    :header-rows: 1
+    :name: supported-input-output-types
+
+    *
+      - Input/Output Types
+      - Library Data Type
+      - AMD Support
+      - CUDA Support
+    *
+      - int8
+      - int8_t
+      - ⚠️
+      - ⚠️
+    *
+      - float8
+      - Not Supported
+      - ❌
+      - ❌
+    *
+      - bfloat8
+      - Not Supported
+      - ❌
+      - ❌
+    *
+      - int16
+      - int16_t
+      - ⚠️
```

Comment:
At rocPRIM PR this is not supported. On AMD platform we are using rocPRIM as backend, so I would expect this will match with rocPRIM page.

Reply:
But int16 is explicitly listed as not supported in the list for rocPRIM. I would expect that means, that the tests failed. Or does "not supported" just mean we don't test for it?

The list should express intent, not the current state of the tests - e.g. if `int16` does currently not pass all tests, but it should be supported, then that's a bug and should be reported.

Reply:
maybe that would be the cleanest. but most of the issues here are with the host side parts of the tests, and not the `hipcub` (or `rocprim`) calls.

Reply:
updated it to reflect the state it should be in